libsolutil/Keccak256.cpp

```diff
@@ -69,25 +69,25 @@ static uint64_t const RC[24] = \
 #define REPEAT6(e) e e e e e e
 #define REPEAT24(e) REPEAT6(e e e e)
 #define REPEAT5(e) e e e e e
-#define FOR5(v, s, e) \
+#define FOR5(type, v, s, e) \
 	v = 0;            \
-	REPEAT5(e; v += s;)
+	REPEAT5(e; v = static_cast<type>(v + s);)
 
 /*** Keccak-f[1600] ***/
 static inline void keccakf(void* state) {
-	uint64_t* a = (uint64_t*)state;
+	auto* a = (uint64_t*)state;
```

Comment:
```suggestion
	auto a = (uint64_t*)state;
```

The `*` is redundant, unless you want to be more explicit!

Reply:
For me either seems fine in this particular case but I like the explicitness, especially if there are more modifiers on the type, e.g. `const*` or `const&`. In think someone was advocating this style on gitter.

Reply:
I am trying to have constness and pointerness of type next to auto, so I would like to leave `auto*`